app/services/data_migrations/create_missing_temp_sites.rb

```diff
@@ -0,0 +1,36 @@
+module DataMigrations
+  class CreateMissingTempSites
+    TIMESTAMP = 20220525161544
+    MANUAL_RUN = true
+
+    def change
+      Site
```

Comment:
instead of finding all sites not attached to course options, we would like to find Site attached to course options that have no corresponding temp site instead. Orphaned sites will be removed when we switch to using temp sites

Reply:
it's looking for sites that _are_ attached to course options (`.where.not(course_options: { id: nil })`).

then line 20 `unless matching_temp_site?(site)` checks for an existing temp site. this is inefficient, since we're iterating through all sites, but i figured the activerecord/sql query to find all sites that don't have a matching tempsite would be pretty complex. meanwhile we only need to run this once, and there aren't so many sites that this is going to hurt the db.

Reply:
my bad I missed the not, your right 👍